<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.4" rel="stylesheet" type="text/css" />


  <link rel="apple-touch-icon" sizes="180x180" href="/assets/apple-touch-icon.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png?v=5.1.4">


  <link rel="mask-icon" href="/assets/safari-pinned-tab.svg?v=5.1.4" color="#222">





  <meta name="keywords" content="生物信息学,翻译文献,RNA-Seq,协议Protocol," />










<meta name="description" content="流程将RNA-seq reads与基因组进行比对 ● 时间&lt;20分钟 将每个样本的reads映射到参考基因组上  123456789101112131415161718192021222324252627282930313233343536$ hisat2 -p 8 --dta -x chrX_data&#x2F;indexes&#x2F;chrX_tran -1chrX_data&#x2F;samples&#x2F;ERR188">
<meta property="og:type" content="article">
<meta property="og:title" content="使用HISAT StringTie和Ballgown进行的转录本表达水平分析（中译）(四)">
<meta property="og:url" content="https://flatig.vip/2022/03/12/%E4%BD%BF%E7%94%A8HISAT%20StringTie%E5%92%8CBallgown%E8%BF%9B%E8%A1%8C%E7%9A%84%E8%BD%AC%E5%BD%95%E6%9C%AC%E8%A1%A8%E8%BE%BE%E6%B0%B4%E5%B9%B3%E5%88%86%E6%9E%90%EF%BC%88%E4%B8%AD%E8%AF%91%EF%BC%89(%E5%9B%9B)/index.html">
<meta property="og:site_name" content="Flatig">
<meta property="og:description" content="流程将RNA-seq reads与基因组进行比对 ● 时间&lt;20分钟 将每个样本的reads映射到参考基因组上  123456789101112131415161718192021222324252627282930313233343536$ hisat2 -p 8 --dta -x chrX_data&#x2F;indexes&#x2F;chrX_tran -1chrX_data&#x2F;samples&#x2F;ERR188">
<meta property="og:locale">
<meta property="article:published_time" content="2022-03-12T09:01:35.000Z">
<meta property="article:modified_time" content="2022-04-16T02:00:29.116Z">
<meta property="article:author" content="flatig_Liu">
<meta property="article:tag" content="生物信息学">
<meta property="article:tag" content="翻译文献">
<meta property="article:tag" content="RNA-Seq">
<meta property="article:tag" content="协议Protocol">
<meta name="twitter:card" content="summary">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '',
    scheme: 'Pisces',
    version: '5.1.4',
    sidebar: {"position":"left","display":"post","offset":12,"b2t":true,"scrollpercent":true,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="https://flatig.vip/2022/03/12/使用HISAT StringTie和Ballgown进行的转录本表达水平分析（中译）(四)/"/>





  <title>使用HISAT StringTie和Ballgown进行的转录本表达水平分析（中译）(四) | Flatig</title>
  








<meta name="generator" content="Hexo 6.2.0"></head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Flatig</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle">我的个人博客站</p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="https://flatig.vip/2022/03/12/%E4%BD%BF%E7%94%A8HISAT%20StringTie%E5%92%8CBallgown%E8%BF%9B%E8%A1%8C%E7%9A%84%E8%BD%AC%E5%BD%95%E6%9C%AC%E8%A1%A8%E8%BE%BE%E6%B0%B4%E5%B9%B3%E5%88%86%E6%9E%90%EF%BC%88%E4%B8%AD%E8%AF%91%EF%BC%89(%E5%9B%9B)/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/images/avatar.gif">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Flatig">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">使用HISAT StringTie和Ballgown进行的转录本表达水平分析（中译）(四)</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2022-03-12T17:01:35+08:00">
                2022-03-12
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E4%B9%A0/" itemprop="url" rel="index">
                    <span itemprop="name">生物信息学习</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <h3 id="流程"><a href="#流程" class="headerlink" title="流程"></a>流程</h3><h5 id="将RNA-seq-reads与基因组进行比对-●-时间-lt-20分钟"><a href="#将RNA-seq-reads与基因组进行比对-●-时间-lt-20分钟" class="headerlink" title="将RNA-seq reads与基因组进行比对 ● 时间&lt;20分钟"></a>将RNA-seq reads与基因组进行比对 ● 时间&lt;20分钟</h5><ol>
<li>将每个样本的reads映射到参考基因组上</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188044_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188044_chrX_2.fastq.gz -S ERR188044_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188104_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188104_chrX_2.fastq.gz -S ERR188104_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1 </span><br><span class="line">chrX_data/samples/ERR188234_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188234_chrX_2.fastq.gz -S ERR188234_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188245_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188245_chrX_2.fastq.gz -S ERR188245_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188257_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188257_chrX_2.fastq.gz -S ERR188257_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188273_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188273_chrX_2.fastq.gz -S ERR188273_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188337_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188337_chrX_2.fastq.gz -S ERR188337_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188383_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188383_chrX_2.fastq.gz -S ERR188383_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188401_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188401_chrX_2.fastq.gz -S ERR188401_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188428_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188428_chrX_2.fastq.gz -S ERR188428_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR188454_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR188454_chrX_2.fastq.gz -S ERR188454_chrX.sam</span><br><span class="line">$ hisat2 -p 8 --dta -x chrX_data/indexes/chrX_tran -1</span><br><span class="line">chrX_data/samples/ERR204916_chrX_1.fastq.gz -2</span><br><span class="line">chrX_data/samples/ERR204916_chrX_2.fastq.gz -S ERR204916_chrX.sam</span><br></pre></td></tr></table></figure>

<ol>
<li>将SAM文件分类并转换为BAM文件。</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">$ samtools sort -@ 8 -o ERR188044_chrX.bam ERR188044_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188104_chrX.bam ERR188104_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188234_chrX.bam ERR188234_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188245_chrX.bam ERR188245_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188257_chrX.bam ERR188257_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188273_chrX.bam ERR188273_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188337_chrX.bam ERR188337_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188383_chrX.bam ERR188383_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188401_chrX.bam ERR188401_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188428_chrX.bam ERR188428_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR188454_chrX.bam ERR188454_chrX.sam</span><br><span class="line">$ samtools sort -@ 8 -o ERR204916_chrX.bam ERR204916_chrX.sam</span><br></pre></td></tr></table></figure>

<blockquote>
<p>  上述命令适用于SAMtools<br>  1.3版或更新的版本。对于旧版本的SAMtools，用户应该参考Box3。</p>
</blockquote>
<p>- Box3 | 使用SAMtools 1.2.1版及以上版本对SAM文件进行分类和转换</p>
<blockquote>
<p>  最近版本的SAMtools可以在一个步骤中把SAM文件转换成BAM文件。对于使用旧版本的人来说，需要两个步骤。如图所示。</p>
</blockquote>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line">Convert SAM files to binary BAM files:</span><br><span class="line"></span><br><span class="line">$ samtools view -bS ERR188044_chrX.sam &gt;ERR188044_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188104_chrX.sam &gt;ERR188104_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188234_chrX.sam &gt;ERR188234_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188245_chrX.sam &gt;ERR188245_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188257_chrX.sam &gt;ERR188257_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188273_chrX.sam &gt;ERR188273_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188337_chrX.sam &gt;ERR188337_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188383_chrX.sam &gt;ERR188383_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188401_chrX.sam &gt;ERR188401_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188428_chrX.sam &gt;ERR188428_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR188454_chrX.sam &gt;ERR188454_chrX_unsorted.bam</span><br><span class="line">$ samtools view -bS ERR204916_chrX.sam &gt;ERR204916_chrX_unsorted.bam</span><br><span class="line"></span><br><span class="line">Sort the BAM files:</span><br><span class="line"></span><br><span class="line">$ samtools sort -@ 8 ERR188044_chrX_unsorted.bam ERR188044_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188104_chrX_unsorted.bam ERR188104_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188234_chrX_unsorted.bam ERR188234_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188245_chrX_unsorted.bam ERR188245_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188257_chrX_unsorted.bam ERR188257_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188273_chrX_unsorted.bam ERR188273_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188337_chrX_unsorted.bam ERR188337_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188383_chrX_unsorted.bam ERR188383_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188401_chrX_unsorted.bam ERR188401_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188428_chrX_unsorted.bam ERR188428_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR188454_chrX_unsorted.bam ERR188454_chrX</span><br><span class="line">$ samtools sort -@ 8 ERR204916_chrX_unsorted.bam ERR204916_chrX</span><br><span class="line"></span><br><span class="line">// Optionally, the user might want to delete temporary files:</span><br><span class="line"></span><br><span class="line">$ rm *.sam *unsorted.bam </span><br></pre></td></tr></table></figure>

<h5 id="组装和量化表达的基因和转录物-●-时间-15分钟"><a href="#组装和量化表达的基因和转录物-●-时间-15分钟" class="headerlink" title="组装和量化表达的基因和转录物 ● 时间~15分钟"></a>组装和量化表达的基因和转录物 ● 时间~15分钟</h5><ol>
<li>为每个样本组装转录本</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188044_chrX.gtf –l ERR188044 ERR188044_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188104_chrX.gtf –l ERR188104 ERR188104_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188234_chrX.gtf –l ERR188234 ERR188234_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188245_chrX.gtf –l ERR188245 ERR188245_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188257_chrX.gtf –l ERR188257 ERR188257_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188273_chrX.gtf –l ERR188273 ERR188273_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188337_chrX.gtf –l ERR188337 ERR188337_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188383_chrX.gtf –l ERR188383 ERR188383_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188401_chrX.gtf –l ERR188401 ERR188401_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188428_chrX.gtf –l ERR188428 ERR188428_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR188454_chrX.gtf –l ERR188454 ERR188454_chrX.bam</span><br><span class="line">$ stringtie -p 8 -G chrX_data/genes/chrX.gtf -o</span><br><span class="line">ERR204916_chrX.gtf –l ERR204916 ERR204916_chrX.bam</span><br></pre></td></tr></table></figure>

<ol>
<li>合并所有样本的转录本</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ stringtie --merge -p 8 -G chrX_data/genes/chrX.gtf -o stringtie_merged.gtf chrX_data/mergelist.txt</span><br></pre></td></tr></table></figure>

<ul>
<li>错误排除</li>
</ul>
<blockquote>
<p>  这里mergelist.txt是一个文本文件，其中有上一步创建的基因转移格式（GTF）文件的名称，每个文件的名称在一行中（参见ChrX_data的例子文件）。如果你不在所有GTF文件所在的同一目录下运行StringTie，那么你可能需要在mergelist.txt中的每个GTF文件名中包括其完整路径。使用任何文本编辑器来创建或编辑（纯文本）mergelist.txt文件。</p>
</blockquote>
<ol>
<li>检查转录本与参考注释的对比情况（可选）</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ gffcompare –r chrX_data/genes/chrX.gtf –G –o merged stringtie_merged.gtf</span><br></pre></td></tr></table></figure>

<p>选项-o指定了gffcompare将创建的输出文件的前缀。上面的命令将生成多个文件，在<em>gffcompare</em>文档中作了解释；更多细节见<strong>BOX1</strong>。</p>
<ol>
<li>估计转录物丰度，并为Ballgown创建表格计数</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188044/ERR188044_chrX.gtf ERR188044_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188104/ERR188104_chrX.gtf ERR188104_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188234/ERR188234_chrX.gtf ERR188234_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188245/ERR188245_chrX.gtf ERR188245_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188257/ERR188257_chrX.gtf ERR188257_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188273/ERR188273_chrX.gtf ERR188273_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188337/ERR188337_chrX.gtf ERR188337_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188383/ERR188383_chrX.gtf ERR188383_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188401/ERR188401_chrX.gtf ERR188401_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188428/ERR188428_chrX.gtf ERR188428_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR188454/ERR188454_chrX.gtf ERR188454_chrX.bam</span><br><span class="line">$ stringtie –e –B -p 8 -G stringtie_merged.gtf -o</span><br><span class="line">ballgown/ERR204916/ERR204916_chrX.gtf ERR204916_chrX.bam</span><br></pre></td></tr></table></figure>

<h5 id="运行差异化表达分析协议-●-时间-5分钟"><a href="#运行差异化表达分析协议-●-时间-5分钟" class="headerlink" title="运行差异化表达分析协议 ● 时间~5分钟"></a>运行差异化表达分析协议 ● 时间~5分钟</h5><ol>
<li>加载相关的R包。这些包包括你将用于执行大部分分析的Ballgown包，以及其他一些有助于这些分析的包，特别是<strong>RSkittleBrewer</strong>（用于设置颜色）、<strong>genefilter</strong>（用于快速计算平均值和变异值）、<strong>dplyr</strong>（用于排序和排列结果）和<strong>devtools</strong>（用于重现性和安装软件包）。</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">$ R</span><br><span class="line">R version 3.2.2 (2015-08-14) -- &quot;Fire Safety&quot;</span><br><span class="line">Copyright (C) 2015 The R Foundation for Statistical Computing</span><br><span class="line">Platform: x86_64-apple-darwin13.4.0 (64-bit)</span><br><span class="line">&gt;library(ballgown)</span><br><span class="line">&gt;library(RSkittleBrewer)</span><br><span class="line">&gt;library(genefilter)</span><br><span class="line">&gt;library(dplyr)</span><br><span class="line">&gt;library(devtools)</span><br></pre></td></tr></table></figure>

<ol>
<li>加载样本的表型数据。一个名为geuvadis_phenodata.csv的示例文件包括在该协议的数据文件中（ChrX_data）。一般来说，你将不得不自己创建这个文件。它包含了关于你的RNA-seq样本的信息，格式如这个csv（逗号分隔的值）文件中所示。文件中的每一行包含每个样本的描述，每一列应该包含一个变量。为了将这个文件读入R，我们使用read.csv命令。在这个文件中，数值是由逗号分隔的，但如果文件是以制表符分隔的，你可以使用函数read.table。</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt;pheno_data = read.csv(&quot;geuvadis_phenodata.csv&quot;)</span><br></pre></td></tr></table></figure>

<ul>
<li>错误排除</li>
</ul>
<ol>
<li>读入由StringTie计算的表达数据。Ballgown目前也支持从Cufflinks6和RSEM17读取数据。要做到这一点，我们使用带有以下三个参数的ballgown命令：存储数据的目录（dataDir，这里简单命名为’Ballgown’），出现在样本名称中的模式（samplePattern）和我们在前一步加载的表型信息（pData）。请注意，一旦创建了Ballgown对象，任何其他Bioconductor32软件包都可以应用于数据分析或数据可视化。这里我们提出了一个标准化的流程，可以用来进行标准的差异表达分析。</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt;bg_chrX = ballgown(dataDir = &quot;ballgown&quot;, samplePattern = &quot;ERR&quot;, pData=pheno_data)</span><br></pre></td></tr></table></figure>

<ul>
<li>错误排除</li>
</ul>
<ol>
<li>使用筛选器去除低丰度基因。RNA-seq数据的一个常见问题是，某些基因经常有非常少的计数或零计数。一个常见的步骤是过滤掉其中的一些。另一种被用于基因表达分析的方法是应用一个方差过滤器。在这里，我们删除所有跨样本方差小于1的转录本。</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt;bg_chrX_filt = subset(bg_chrX,&quot;rowVars(texpr(bg_chrX)) &gt;1&quot;,genomesubset=TRUE)</span><br></pre></td></tr></table></figure>

<ol>
<li>识别在统计学上显示组间差异的转录本。我们可能想做的一件事是确保我们考虑到其他变量引起的表达变化。作为一个例子，我们将寻找在性别间有差异表达的转录本，同时校正由于群体变量造成的任何表达差异。我们可以使用Ballgown的stattest函数来做这件事。我们设置了<strong>getFC&#x3D;TRUE</strong>参数，这样我们就可以查看两组之间经混杂因素调整后的折叠变化。<br>请注意，Ballgown的统计测试是一个基于标准线性模型的比较。对于小样本量（每组n&lt;4），通常最好进行正则化。这可以用<em>Bioconductor</em>中的limma包来完成。其他正则化方法，如DESeq和edgeR可以应用于基因或外显子计数，但它们不适合直接应用于FPKM丰度估计。统计测试使用累积的上四分位数规范化(<em>cumulative<br>upper quartile normalization</em>)。</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">&gt;results_transcripts = stattest(bg_chrX_filt,</span><br><span class="line">feature=&quot;transcript&quot;,covariate=&quot;sex&quot;,adjustvars = c(&quot;population&quot;), </span><br><span class="line">getFC=TRUE, meas=&quot;FPKM&quot;)</span><br></pre></td></tr></table></figure>

<ol>
<li>识别组间有统计学差异的基因。为此，我们可以运行与识别差异表达的转录本相同的函数，在这里我们在stattest命令中设置feature&#x3D;”gene”</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">&gt;results_genes = stattest(bg_chrX_filt, feature=&quot;gene&quot;,</span><br><span class="line">covariate=&quot;sex&quot;, adjustvars = c(&quot;population&quot;), getFC=TRUE,</span><br><span class="line">meas=&quot;FPKM&quot;)</span><br></pre></td></tr></table></figure>

<ol>
<li>在results_transcripts数据框中添加基因名称和基因ID</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">&gt;results_transcripts =</span><br><span class="line">data.frame(geneNames=ballgown::geneNames(bg_chrX_filt),</span><br><span class="line">geneIDs=ballgown::geneIDs(bg_chrX_filt), results_transcripts)</span><br></pre></td></tr></table></figure>

<ol>
<li>将结果从最小的P值排序到最大</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;results_transcripts = arrange(results_transcripts,pval)</span><br><span class="line">&gt;results_genes = arrange(results_genes,pval)</span><br></pre></td></tr></table></figure>

<ul>
<li>两性之间差异表达的转录物（q值&lt;5%） 基因名|基因ID|转录本名|ID|Fold<br>Change|P值|q值 :–:|:–:|:–:|:–:|:–:|:–:|:–: XIST<br>MSTRG.506|NR_001564|1729|0.003255|7.0447e-10|1.6160e-06<br>|MSTRG.506|MSTRG.506.1|1728|0.016396|1.2501e-08|1.4339e-05<br>TSIX|MSTRG.505|NR_003255|1726|0.083758|2.4939e-06|1.9070e-03<br>|MSTRG.506|MSTRG.506.2|1727|0.047965|3.7175e-06|2.1319e-03<br>|MSTRG.585|MSTRG.585.1|1919|7.318925|9.3945e-06|3.7715e-03<br>PNPLA4|MSTRG.56|NM_004650|203|0.466647|9.8645e-06|3.7715e-03<br>|MSTRG.506|MSTRG.506.5|1731|0.046993|2.1350e-05|6.9968e-03<br>|MSTRG.592|MSTRG.592.1|1923|9.186257|3.5077e-05|1.0058e-02<br>|MSTRG.518|MSTRG.518.1|1744|11.972859|4.4476e-05|1.1336e-02</li>
</ul>
<blockquote>
<p>  Fold<br>  Change是指女性与男性的表达比例；因此，低于1的数值意味着转录物在男性中的表达水平较低。</p>
</blockquote>
<ol>
<li>将结果写入一个可以共享和分发的csv文件</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">&gt;write.csv(results_transcripts, &quot;chrX_transcript_results.csv&quot;,</span><br><span class="line">row.names=FALSE)</span><br><span class="line">&gt;write.csv(results_genes, &quot;chrX_gene_results.csv&quot;,</span><br><span class="line">row.names=FALSE) </span><br></pre></td></tr></table></figure>

<ol>
<li>识别q值&lt;0.05的转录物和基因</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;subset(results_transcripts,results_transcripts$qval&lt;0.05)</span><br><span class="line">&gt;subset(results_genes,results_genes$qval&lt;0.05)</span><br></pre></td></tr></table></figure>

<p>每个命令的Ballgown输出都会出现在屏幕上；我们在下面的<strong>Table3</strong>（转录物）和<strong>Table4</strong>（基因）中显示了结果。如表所示，X号染色体有9个转录本在两性之间有差异表达（使用q值阈值为0.05），其中3个对应于已知基因的异构体（XIST、TSIX和PNPLA4）。在基因水平上（表4），X染色体在相同的q值临界点上有10个差异表达的基因。(如果使用不同版本的程序，这些表格可能包含略有不同的结果）。</p>
<ul>
<li><p>错误排除</p>
</li>
<li><p>错误排除</p>
</li>
<li><p>错误排除</p>
</li>
</ul>

      
    </div>
    
    
    

    
      <div>
        <div id="wechat_subscriber" style="display: block; padding: 10px 0; margin: 20px auto; width: 100%; text-align: center">
    <img id="wechat_subscriber_qcode" src="https://pic.imgdb.cn/item/625cc849239250f7c53b2990.png" alt=" wechat" style="width: 200px; max-width: 100%;"/>
    <div>欢迎大家订阅我的微信公众号！</div>
</div>

      </div>
    

    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/" rel="tag"># 生物信息学</a>
          
            <a href="/tags/%E7%BF%BB%E8%AF%91%E6%96%87%E7%8C%AE/" rel="tag"># 翻译文献</a>
          
            <a href="/tags/RNA-Seq/" rel="tag"># RNA-Seq</a>
          
            <a href="/tags/%E5%8D%8F%E8%AE%AEProtocol/" rel="tag"># 协议Protocol</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2022/03/12/%E4%BD%BF%E7%94%A8HISAT%20StringTie%E5%92%8CBallgown%E8%BF%9B%E8%A1%8C%E7%9A%84%E8%BD%AC%E5%BD%95%E6%9C%AC%E8%A1%A8%E8%BE%BE%E6%B0%B4%E5%B9%B3%E5%88%86%E6%9E%90%EF%BC%88%E4%B8%AD%E8%AF%91%EF%BC%89(%E4%B8%89)/" rel="next" title="使用HISAT StringTie和Ballgown进行的转录本表达水平分析（中译）(三)">
                <i class="fa fa-chevron-left"></i> 使用HISAT StringTie和Ballgown进行的转录本表达水平分析（中译）(三)
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2022/03/14/%E8%AF%BA%E8%B4%9D%E5%B0%94%E6%96%87%E5%AD%A6%E5%A5%96%E5%BE%97%E4%B8%BB%E7%BB%8F%E5%85%B8%E6%83%85%E8%AF%97%E3%80%8A%E5%A6%82%E6%9E%9C%E4%BD%A0%E5%BF%98%E8%AE%B0%E6%88%91%E3%80%8B/" rel="prev" title="如果你忘记我">
                如果你忘记我 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          

  



        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview-wrap">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <p class="site-author-name" itemprop="name"></p>
              <p class="site-description motion-element" itemprop="description"></p>
          </div>

          <nav class="site-state motion-element">

            
              <div class="site-state-item site-state-posts">
              
                <a href="/archives/%7C%7C%20archive">
              
                  <span class="site-state-item-count">14</span>
                  <span class="site-state-item-name">日志</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-categories">
                <a href="/categories/index.html">
                  <span class="site-state-item-count">7</span>
                  <span class="site-state-item-name">分类</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-tags">
                <a href="/tags/index.html">
                  <span class="site-state-item-count">17</span>
                  <span class="site-state-item-name">标签</span>
                </a>
              </div>
            

          </nav>

          

          
            <div class="links-of-author motion-element">
                
                  <span class="links-of-author-item">
                    <a href="https://github.com/Flatigers" target="_blank" title="GitHub">
                      
                        <i class="fa fa-fw fa-github"></i>GitHub</a>
                  </span>
                
                  <span class="links-of-author-item">
                    <a href="flatig@126.com" target="_blank" title="E-Mail">
                      
                        <i class="fa fa-fw fa-envelope"></i>E-Mail</a>
                  </span>
                
            </div>
          

          
          

          
          

          

        </div>
      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%B5%81%E7%A8%8B"><span class="nav-number">1.</span> <span class="nav-text">流程</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#%E5%B0%86RNA-seq-reads%E4%B8%8E%E5%9F%BA%E5%9B%A0%E7%BB%84%E8%BF%9B%E8%A1%8C%E6%AF%94%E5%AF%B9-%E2%97%8F-%E6%97%B6%E9%97%B4-lt-20%E5%88%86%E9%92%9F"><span class="nav-number">1.0.1.</span> <span class="nav-text">将RNA-seq reads与基因组进行比对 ● 时间&lt;20分钟</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E5%92%8C%E9%87%8F%E5%8C%96%E8%A1%A8%E8%BE%BE%E7%9A%84%E5%9F%BA%E5%9B%A0%E5%92%8C%E8%BD%AC%E5%BD%95%E7%89%A9-%E2%97%8F-%E6%97%B6%E9%97%B4-15%E5%88%86%E9%92%9F"><span class="nav-number">1.0.2.</span> <span class="nav-text">组装和量化表达的基因和转录物 ● 时间~15分钟</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#%E8%BF%90%E8%A1%8C%E5%B7%AE%E5%BC%82%E5%8C%96%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90%E5%8D%8F%E8%AE%AE-%E2%97%8F-%E6%97%B6%E9%97%B4-5%E5%88%86%E9%92%9F"><span class="nav-number">1.0.3.</span> <span class="nav-text">运行差异化表达分析协议 ● 时间~5分钟</span></a></li></ol></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      
        <div class="back-to-top">
          <i class="fa fa-arrow-up"></i>
          
            <span id="scrollpercent"><span>0</span>%</span>
          
        </div>
      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">flatig_Liu</span>

  
</div>


  <div class="powered-by">由 <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a> 强力驱动</div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">主题 &mdash; <a class="theme-link" target="_blank" href="https://github.com/iissnan/hexo-theme-next">NexT.Pisces</a> v5.1.4</div>




        







        
      </div>
    </footer>

    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>
  

  
  
    <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.4"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.4"></script>



  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.4"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.4"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.4"></script>



  


  




	





  





  












  





  

  

  

  
  

  

  

  

</body>
</html>
