<!DOCTYPE html>
<html itemscope itemtype="http://schema.org/WebPage" lang="zh-CN">
  <head>
    
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
    <meta name="robots" content="noodp" />
    <title>运行一个RNA-Seq流程 - flatig-我的个人博客站</title><meta name="author" content="flatig">
<meta name="author-link" content="https://github.com/Flatigers">
<meta name="description" content="运行一个RNA-Seq流程 此流程根据《使用HISAT StringTie和Ballgown进行的转录本表达水平分析》重复文中的RNA-Seq实验 流程设计 本流程分为如下步骤： 生物信息服务器的搭建 生信环境的配置 在生信环境上操作RNA-seq实验 统计数据及结果分析 生信服务器的搭建 生物信息学分析所使用的数据量" /><meta name="keywords" content='生物信息学, RNA-Seq' /><meta itemprop="name" content="运行一个RNA-Seq流程">
<meta itemprop="description" content="运行一个RNA-Seq流程 此流程根据《使用HISAT StringTie和Ballgown进行的转录本表达水平分析》重复文中的RNA-Seq实验 流程设计 本流程分为如下步骤： 生物信息服务器的搭建 生信环境的配置 在生信环境上操作RNA-seq实验 统计数据及结果分析 生信服务器的搭建 生物信息学分析所使用的数据量"><meta itemprop="datePublished" content="2022-04-18T10:20:02+00:00" />
<meta itemprop="dateModified" content="2022-04-18T10:20:02+00:00" />
<meta itemprop="wordCount" content="2305"><meta itemprop="image" content="https://flatig.vip/images/logo.png"/>
<meta itemprop="keywords" content="生物信息学,RNA-Seq," /><meta property="og:title" content="运行一个RNA-Seq流程" />
<meta property="og:description" content="运行一个RNA-Seq流程 此流程根据《使用HISAT StringTie和Ballgown进行的转录本表达水平分析》重复文中的RNA-Seq实验 流程设计 本流程分为如下步骤： 生物信息服务器的搭建 生信环境的配置 在生信环境上操作RNA-seq实验 统计数据及结果分析 生信服务器的搭建 生物信息学分析所使用的数据量" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/" /><meta property="og:image" content="https://flatig.vip/images/logo.png"/><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2022-04-18T10:20:02+00:00" />
<meta property="article:modified_time" content="2022-04-18T10:20:02+00:00" />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="https://flatig.vip/images/logo.png"/>

<meta name="twitter:title" content="运行一个RNA-Seq流程"/>
<meta name="twitter:description" content="运行一个RNA-Seq流程 此流程根据《使用HISAT StringTie和Ballgown进行的转录本表达水平分析》重复文中的RNA-Seq实验 流程设计 本流程分为如下步骤： 生物信息服务器的搭建 生信环境的配置 在生信环境上操作RNA-seq实验 统计数据及结果分析 生信服务器的搭建 生物信息学分析所使用的数据量"/>
<meta name="application-name" content="Flatig 博客站">
<meta name="apple-mobile-web-app-title" content="Flatig 博客站"><meta name="theme-color" data-light="#ffffff" data-dark="#252627" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
    <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/" /><link rel="prev" href="https://flatig.vip/posts/%E4%B8%AA%E4%BA%BA%E5%85%AC%E4%BC%97%E5%8F%B7%E5%BC%80%E9%80%9A%E5%95%A6/" /><link rel="next" href="https://flatig.vip/posts/%E5%AE%89%E8%A3%85%E4%BD%BF%E7%94%A8%E6%9B%B4%E5%8F%AF%E9%9D%A0%E7%9A%84linux%E5%8F%91%E8%A1%8C%E7%89%88debian/" /><link rel="stylesheet" href="/css/style.min.css"><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"><link rel="stylesheet" href="/lib/animate/animate.min.css"><script type="application/ld+json">
  {
    "@context": "http://schema.org",
    "@type": "BlogPosting",
    "headline": "运行一个RNA-Seq流程",
    "inLanguage": "zh-CN",
    "mainEntityOfPage": {
      "@type": "WebPage",
      "@id": "https:\/\/flatig.vip\/posts\/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B\/"
    },"genre": "posts","keywords": "生物信息学, RNA-Seq","wordcount":  2305 ,
    "url": "https:\/\/flatig.vip\/posts\/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B\/","datePublished": "2022-04-18T10:20:02+00:00","dateModified": "2022-04-18T10:20:02+00:00","publisher": {
      "@type": "Organization",
      "name": "flatig","logo": "https:\/\/flatig.vip\/images\/logo.png"},"author": {
        "@type": "Person",
        "name": "flatig"
      },"description": ""
  }
  </script></head>
  <body data-header-desktop="sticky" data-header-mobile="auto"><script>(window.localStorage?.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('auto' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : 'auto' === 'dark')) && document.body.setAttribute('data-theme', 'dark');</script><div class="wrapper" data-page-style="normal"><header class="desktop animate__faster" id="header-desktop">
  <div class="header-wrapper" data-github-corner="right">
    <div class="header-title">
      <a href="/" title="flatig-我的个人博客站"><img loading="lazy" src="/images/logo.svg" srcset="/images/logo.svg, /images/logo.svg 1.5x, /images/logo.svg 2x" sizes="auto" data-title="flatig-我的个人博客站" data-alt="flatig-我的个人博客站" class="logo" style="background: url(/svg/loading.min.svg) no-repeat center;" onload="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}this.dataset.lazyloaded='';" onerror="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}"/><span class="header-title-text">Flatig</span></a><span id="typeit-header-subtitle-desktop" class="typeit header-subtitle"></span></div>
    <nav>
      <ul class="menu"><li class="menu-item">
              <a
                class="menu-link"
                href="/posts/"
                
                
              ><i class="fa-solid fa-archive fa-fw fa-sm" aria-hidden="true"></i> 文章</a></li><li class="menu-item">
              <a
                class="menu-link"
                href="/categories/"
                
                
              ><i class="fa-solid fa-th fa-fw fa-sm" aria-hidden="true"></i> 分类</a></li><li class="menu-item">
              <a
                class="menu-link"
                href="/tags/"
                
                
              ><i class="fa-solid fa-tags fa-fw fa-sm" aria-hidden="true"></i> 标签</a></li><li class="menu-item">
              <a
                class="menu-link"
                href="/tags/"
                
                
              ><i class="fa-solid fa-book fa-fw fa-sm" aria-hidden="true"></i> 文档</a></li><li class="menu-item">
              <a
                class="menu-link"
                href="/friends/"
                
                
              ><i class="fa-solid fa-users fa-fw fa-sm" aria-hidden="true"></i> 友链</a></li><li class="menu-item">
              <a
                class="menu-link"
                href="/about/"
                
                
              ><i class="fa-solid fa-user-tie fa-fw fa-sm" aria-hidden="true"></i> 关于</a></li><li class="menu-item delimiter"></li><li class="menu-item search" id="search-desktop">
            <input type="text" placeholder="Search something" id="search-input-desktop">
            <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
              <i class="fa-solid fa-search fa-fw" aria-hidden="true"></i>
            </a>
            <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
              <i class="fa-solid fa-times-circle fa-fw" aria-hidden="true"></i>
            </a>
            <span class="search-button search-loading" id="search-loading-desktop">
              <i class="fa-solid fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
            </span>
          </li><li class="menu-item theme-switch" title="切换主题">
          <i class="fa-solid fa-adjust fa-fw" aria-hidden="true"></i>
        </li></ul>
    </nav>
  </div>
</header><header class="mobile animate__faster" id="header-mobile">
  <div class="header-container">
    <div class="header-wrapper">
      <div class="header-title">
        <a href="/" title="flatig-我的个人博客站"><img loading="lazy" src="/images/logo.svg" srcset="/images/logo.svg, /images/logo.svg 1.5x, /images/logo.svg 2x" sizes="auto" data-title="/images/logo.svg" data-alt="/images/logo.svg" class="logo" style="background: url(/svg/loading.min.svg) no-repeat center;" onload="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}this.dataset.lazyloaded='';" onerror="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}"/><span class="header-title-text">Flatig</span></a><span id="typeit-header-subtitle-mobile" class="typeit header-subtitle"></span></div>
      <div class="menu-toggle" id="menu-toggle-mobile">
        <span></span><span></span><span></span>
      </div>
    </div>
    <nav>
      <ul class="menu" id="menu-mobile"><li class="search-wrapper">
            <div class="search mobile" id="search-mobile">
              <input type="text" placeholder="Search something" id="search-input-mobile">
              <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                <i class="fa-solid fa-search fa-fw" aria-hidden="true"></i>
              </a>
              <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                <i class="fa-solid fa-times-circle fa-fw" aria-hidden="true"></i>
              </a>
              <span class="search-button search-loading" id="search-loading-mobile">
                <i class="fa-solid fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
              </span>
            </div>
            <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
              取消
            </a>
          </li><li
              class="menu-item"
            ><a
                  class="menu-link"
                  href="/posts/"
                  
                  
                ><i class="fa-solid fa-archive fa-fw fa-sm" aria-hidden="true"></i> 文章</a></li><li
              class="menu-item"
            ><a
                  class="menu-link"
                  href="/categories/"
                  
                  
                ><i class="fa-solid fa-th fa-fw fa-sm" aria-hidden="true"></i> 分类</a></li><li
              class="menu-item"
            ><a
                  class="menu-link"
                  href="/tags/"
                  
                  
                ><i class="fa-solid fa-tags fa-fw fa-sm" aria-hidden="true"></i> 标签</a></li><li
              class="menu-item"
            ><a
                  class="menu-link"
                  href="/tags/"
                  
                  
                ><i class="fa-solid fa-book fa-fw fa-sm" aria-hidden="true"></i> 文档</a></li><li
              class="menu-item"
            ><a
                  class="menu-link"
                  href="/friends/"
                  
                  
                ><i class="fa-solid fa-users fa-fw fa-sm" aria-hidden="true"></i> 友链</a></li><li
              class="menu-item"
            ><a
                  class="menu-link"
                  href="/about/"
                  
                  
                ><i class="fa-solid fa-user-tie fa-fw fa-sm" aria-hidden="true"></i> 关于</a></li><li
              class="menu-item text-center"
            ><a
                  class="menu-link"
                  href="https://github.com/Flatigers"
                  title="GitHub"
                  rel="noopener noreferrer" target="_blank"
                ><i class='fa-brands fa-github fa-fw' aria-hidden='true'></i> </a></li><li class="menu-item menu-system">
          <span class="menu-system-item theme-switch" title="切换主题"><i class="fa-solid fa-adjust fa-fw" aria-hidden="true"></i></span></li>
      </ul>
    </nav>
  </div>
</header><div class="search-dropdown desktop">
    <div id="search-dropdown-desktop"></div>
  </div>
  <div class="search-dropdown mobile">
    <div id="search-dropdown-mobile"></div>
  </div><nav aria-label="breadcrumb" class="breadcrumb-container sticky">
    <ol class="breadcrumb"><li class="breadcrumb-item"><a href="/posts/" title="Posts">Posts</a></li><li class="breadcrumb-item active" aria-current="page">运行一个RNA-Seq流程</li>
    </ol>
  </nav><main class="container"><aside class="toc" id="toc-auto"><h2 class="toc-title">目录&nbsp;<i class="toc-icon fa-solid fa-angle-down fa-fw" aria-hidden="true"></i></h2>
      <div class="toc-content" id="toc-content-auto"></div></aside>

  <aside class="aside-custom">
    </aside>

  <article class="page single">
    <div class="header"><h1 class="single-title animate__animated animate__flipInX"><span>运行一个RNA-Seq流程</span>
      </h1></div><div class="post-meta">
      <div class="post-meta-line"><span class="post-author"><a href="https://github.com/Flatigers" title="作者"target="_blank" rel="external nofollow noopener noreferrer author" class="author"><img loading="lazy" src="https://gravatar.loli.net/avatar/73af57b4975d7b4b5c5067f9a84805b3?s=32&amp;d=mp" srcset="https://gravatar.loli.net/avatar/73af57b4975d7b4b5c5067f9a84805b3?s=32&amp;d=mp, https://gravatar.loli.net/avatar/73af57b4975d7b4b5c5067f9a84805b3?s=32&amp;d=mp 1.5x, https://gravatar.loli.net/avatar/73af57b4975d7b4b5c5067f9a84805b3?s=32&amp;d=mp 2x" sizes="auto" data-title="flatig" data-alt="flatig" class="avatar" style="background: url(/svg/loading.min.svg) no-repeat center;" onload="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}this.dataset.lazyloaded='';" onerror="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}"/>&nbsp;flatig</a></span>
          <span class="post-category">收录于 <a href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E4%B9%A0/"><i class="fa-regular fa-folder fa-fw" aria-hidden="true"></i> 生物信息学习</a></span></div>
      <div class="post-meta-line"><span title="发布于 2022-04-18 10:20:02"><i class="fa-regular fa-calendar-alt fa-fw me-1" aria-hidden="true"></i><time datetime="2022-04-18">2022-04-18</time></span>&nbsp;<span title="更新于 2022-04-18 10:20:02"><i class="fa-regular fa-edit fa-fw me-1" aria-hidden="true"></i><time datetime="2022-04-18">2022-04-18</time></span>&nbsp;<span title="2305 字"><i class="fa-solid fa-pencil-alt fa-fw me-1" aria-hidden="true"></i>约 2400 字</span>&nbsp;<span><i class="fa-regular fa-clock fa-fw me-1" aria-hidden="true"></i>预计阅读 5 分钟</span>&nbsp;</div>
    </div><div class="details toc" id="toc-static" data-kept="false">
        <div class="details-summary toc-title">
          <span>目录</span>
          <span><i class="details-icon fa-solid fa-angle-right" aria-hidden="true"></i></span>
        </div>
        <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#运行一个rna-seq流程">运行一个RNA-Seq流程</a>
      <ul>
        <li><a href="#流程设计">流程设计</a></li>
        <li><a href="#生信服务器的搭建">生信服务器的搭建</a></li>
        <li><a href="#生信环境的配置">生信环境的配置</a></li>
        <li><a href="#在生信环境上操作rna-seq实验">在生信环境上操作RNA-seq实验</a></li>
        <li><a href="#统计数据及结果分析">统计数据及结果分析</a></li>
      </ul>
    </li>
  </ul>
</nav></div>
      </div><div class="content" id="content" data-end-flag="-- ~ --"><div class="details admonition warning open">
      <div class="details-summary admonition-title">
        <i class="icon fa-solid fa-exclamation-triangle fa-fw" aria-hidden="true"></i>警告<i class="details-icon fa-solid fa-angle-right fa-fw" aria-hidden="true"></i>
      </div>
      <div class="details-content">
        <div class="admonition-content">本文最后更新于 2022-04-18，文中内容可能已过时。</div>
      </div>
    </div><h2 id="运行一个rna-seq流程">运行一个RNA-Seq流程</h2>
<hr>
<blockquote>
<p>此流程根据《使用HISAT StringTie和Ballgown进行的转录本表达水平分析》重复文中的RNA-Seq实验</p>
</blockquote>
<h3 id="流程设计">流程设计</h3>
<p>本流程分为如下步骤：</p>
<ol>
<li>生物信息服务器的搭建</li>
<li>生信环境的配置</li>
<li>在生信环境上操作RNA-seq实验</li>
<li>统计数据及结果分析</li>
</ol>
<h3 id="生信服务器的搭建">生信服务器的搭建</h3>
<p>生物信息学分析所使用的数据量极大，往往需要数百G以上的存储空间及大量内存，这样的需求在一般PC上较难实现。而且生信分析往往需要众人分工协作，由此搭建一个公用服务器十分重要。</p>
<p>本次重复的实验使用的数据较小，在个人PC上完全可以操作，需要的硬盘存储空间约<strong>20G</strong>以内，由于条件的限制，作者本次实验在个人Linux系统上进行操作。</p>
<p>生物信息学实验使用服务器是必不可少的，在今后的操作中，读者需熟练掌握Linux操作系统及shell命令编程的使用，熟练使用云Linux服务器进行操作。国内云计算提供商有<strong>腾讯云，阿里云，华为云</strong>等，新用户均有数月的云服务器体验，读者可根据自己的实际情况自行选择。</p>
<h3 id="生信环境的配置">生信环境的配置</h3>
<h4 id="系统环境">系统环境</h4>
<p>根据此流程需要进行的操作，本次实验需要在Linux系统上配置的环境如下：</p>
<ul>
<li>R语言环境</li>
<li>R包：alyssafrazee/RSkittleBrewer, <strong>ballgown</strong>, genefilter, dplyr, devtools</li>
<li>分析软件：Hisat2, StringTie, <strong>Samtools</strong>， gffcompare</li>
</ul>
<p>安装hisat2和StringTie:</p>
<div class="highlight" id="id-1"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">sudo apt update
</span></span><span class="line"><span class="cl">sudo apt install hisat2
</span></span><span class="line"><span class="cl">sudo apt install stringtie</span></span></code></pre></td></tr></table>
</div>
</div><p>安装Conda并使用其安装Samtools, gffcompare:</p>
<div class="highlight" id="id-2"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">wget -c https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
</span></span><span class="line"><span class="cl"># 适合于linux 64位
</span></span><span class="line"><span class="cl">chmod 777 Miniconda3-latest-Linux-x86_64.sh
</span></span><span class="line"><span class="cl">#给执行权限
</span></span><span class="line"><span class="cl">bash Miniconda3-latest-Linux-x86_64.sh
</span></span><span class="line"><span class="cl">#运行安装程序，所有选项均选&#39;yes&#39;
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/
</span></span><span class="line"><span class="cl">conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/
</span></span><span class="line"><span class="cl">conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge/
</span></span><span class="line"><span class="cl">conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/bioconda/
</span></span><span class="line"><span class="cl">#配置conda频道，这里选择清华Tuna镜像
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">conda install samtools
</span></span><span class="line"><span class="cl">conda install gffcompare
</span></span><span class="line"><span class="cl">#使用conda安装Samtools和gffcompare
</span></span><span class="line"><span class="cl">#这里使用conda安装是因为作者使用apt安装samtools时出现问题
</span></span><span class="line"><span class="cl">#conda是专用于生物信息学分析的环境配置器
</span></span><span class="line"><span class="cl">#Hisat, StringTie以及其他R包均可使用conda安装</span></span></code></pre></td></tr></table>
</div>
</div><p>安装 R 和 R包Ballgown：</p>
<div class="highlight" id="id-3"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">conda create -n Rspcae
</span></span><span class="line"><span class="cl">#创建名为Rspace的conda环境，可以替换成自己需要的名称
</span></span><span class="line"><span class="cl">source activate Rspace
</span></span><span class="line"><span class="cl">#激活此环境
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">conda install r-base=4.1.3
</span></span><span class="line"><span class="cl">#安装4.1.3的R，可以在conda官网包管理页面查询r包的最新版本
</span></span><span class="line"><span class="cl">conda install r-ballgown
</span></span><span class="line"><span class="cl">#conda 安装r包需要在包名前 添加 r-
</span></span><span class="line"><span class="cl">#此处作者使用conda安装r包，也是因为在R中安装遇到问题
</span></span><span class="line"><span class="cl">#若无问题，可以直接在R中安装任何R包，具体为：
</span></span><span class="line"><span class="cl">#install.packages(ballgown) 或
</span></span><span class="line"><span class="cl">#BiocManager::install(&#34;ballgown&#34;)
</span></span><span class="line"><span class="cl">#使用BiocManager需先安装，参加下一步
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">conda deactivate
</span></span><span class="line"><span class="cl">#安装完成后退出当前conda环境</span></span></code></pre></td></tr></table>
</div>
</div><p>安装其他R包：</p>
<div class="highlight" id="id-4"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">R        
</span></span><span class="line"><span class="cl">#进入R环境
</span></span><span class="line"><span class="cl">install.packages(“devtools”)
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">install.packages(&#34;BiocManager&#34;)
</span></span><span class="line"><span class="cl">library(&#34;BiocManager&#34;)
</span></span><span class="line"><span class="cl">#安装并导入BiocManager
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">BiocManager::install(c(&#34;alyssafrazee/RSkittleBrewer&#34;,&#34;ballgown&#34;,&#34;genefilter&#34;,&#34;dplyr&#34;))
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#source(&#34;http://www.bioconductor.org/biocLite.R&#34;)
</span></span><span class="line"><span class="cl">#BiocManager::biocLite(c(&#34;alyssafrazee/RSkittleBrewer&#34;,&#34;ballgown&#34;,&#34;genefilter&#34;,&#34;dplyr&#34;))
</span></span><span class="line"><span class="cl">#安装R包,此项为旧版本用法
</span></span><span class="line"><span class="cl">#此处也可使用：
</span></span><span class="line"><span class="cl">#install.packages()</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="资源">资源</h4>
<p>此处所需资源为文章中提供的测序数据及参考序列等资料：</p>
<div class="highlight" id="id-5"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">cd /home/username/RNA
</span></span><span class="line"><span class="cl">nohup wget ftp://ftp.ccb.jhu.edu/pub/RNAseq_protocol/chrX_data.tar.gz &amp;
</span></span><span class="line"><span class="cl">tar zxvf chrX_data.tar.gz
</span></span><span class="line"><span class="cl">#先cd到工作环境下文件夹
</span></span><span class="line"><span class="cl">#下载并解压</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="在生信环境上操作rna-seq实验">在生信环境上操作RNA-seq实验</h3>
<h4 id="hisat2将reads与genome匹配">hisat2:将reads与genome匹配</h4>
<div class="highlight" id="id-6"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">for i in *1.fastq.gz; 
</span></span><span class="line"><span class="cl">do
</span></span><span class="line"><span class="cl">i=${i%1.fastq.gz*}; 
</span></span><span class="line"><span class="cl">nohup hisat2 -p 8 --dta -x /home/username/RNA/chrX_data/indexes/chrX_tran -1 ${i}1.fastq.gz -2 ${i}2.fastq.gz -S /home/username/RNA/chrX_data/result/${i}align.sam /home/username/RNA/chrX_data/result/${i}align.log &amp; 
</span></span><span class="line"><span class="cl">done
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#此步骤将在/ home/username/RNA/chrX_data/result/ 下生成sample reads的sam文件</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="samtools将sam文件转换为bam文件">samtools:将sam文件转换为bam文件</h4>
<div class="highlight" id="id-7"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">for i in *.sam;
</span></span><span class="line"><span class="cl">do 
</span></span><span class="line"><span class="cl">i=${i%_align.sam*};
</span></span><span class="line"><span class="cl">nohup samtools sort -@ 8 -o ${i}.bam ${i}_align.sam &amp; 
</span></span><span class="line"><span class="cl">done
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#此步骤将在同目录下生成bam文件</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="stringtie组装转录本并定量表达基因">stringtie:组装转录本并定量表达基因</h4>
<div class="highlight" id="id-8"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">for i in *.bam; 
</span></span><span class="line"><span class="cl">do 
</span></span><span class="line"><span class="cl">i=${i%.bam*};
</span></span><span class="line"><span class="cl">nohup stringtie -p 8 -G /home/username/RNA/chrX_data/genes/chrX.gtf -o ${i}.gtf ${i}.bam &amp; 
</span></span><span class="line"><span class="cl">done
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#此步骤将在同目录下生成gtf文件</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="stringtie合并所有转录本">stringtie:合并所有转录本</h4>
<div class="highlight" id="id-9"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">stringtie --merge -p 8 -G  /home/username/RNA/chrX_data/genes/chrX.gtf -o stringtie_merged.gtf  /home/username/RNA/chrX_data/mergelist.txt
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#此步骤将在同目录下生成 stringtie_merged.gtf 文件</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="gffcompare检测相对于注释文件转录本的组装情况">gffcompare:检测相对于注释文件，转录本的组装情况</h4>
<div class="highlight" id="id-10"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">gffcompare -r /home/username/RNA/chrX_data/genes/chrX.gtf  -G -o merged stringtie_merged.gtf
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#此步骤将生成如下文件：
</span></span><span class="line"><span class="cl">#gffcmp.annotated.gtf
</span></span><span class="line"><span class="cl">#它在每个转录本上添加一个 “类代码”（描述参见流程文档）和注释文件中的转录本名称
</span></span><span class="line"><span class="cl">#gffcmp.stats
</span></span><span class="line"><span class="cl">#输出文件中计算不同基因特征的灵敏度和精确度（P值）统计</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="stringtie重新组装转录本并估算基因表达丰度并为ballgown创建读入文件">stringtie:重新组装转录本并估算基因表达丰度，并为ballgown创建读入文件</h4>
<div class="highlight" id="id-11"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">for i in *_chrX.bam; 
</span></span><span class="line"><span class="cl">do 
</span></span><span class="line"><span class="cl">i=${i%_chrX.bam*};
</span></span><span class="line"><span class="cl">nohup stringtie -e -B -p 8 -G stringtie_merged.gtf -o ballgown/${i}/${i}_chrX.gtf  ${i}_chrX.bam &amp; 
</span></span><span class="line"><span class="cl">done
</span></span><span class="line"><span class="cl">#此步骤将在 /home/username/RNA/chrX_data/ballgown/ 下生成
</span></span><span class="line"><span class="cl">#以sample名为文件夹名的的ballgown读取文件</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="统计数据及结果分析">统计数据及结果分析</h3>
<p>此步骤将在R环境中进行统计步骤及结果分析</p>
<h4 id="数据处理">数据处理</h4>
<ul>
<li>进入R并加载R包</li>
</ul>
<div class="highlight" id="id-12"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">library(ballgown)
</span></span><span class="line"><span class="cl">library(RSkittleBrewer)
</span></span><span class="line"><span class="cl">library(genefilter)
</span></span><span class="line"><span class="cl">library(dplyr)
</span></span><span class="line"><span class="cl">library(devtools)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>加载表型数据</li>
</ul>
<div class="highlight" id="id-13"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">setwd(&#34;/home/username/RNA/chrX_data/&#34;)
</span></span><span class="line"><span class="cl">#设置R工作环境目录
</span></span><span class="line"><span class="cl">pheno_data = read.csv(&#34;geuvadis_phenodata.csv&#34;)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>读入StringTie的计算结果</li>
</ul>
<div class="highlight" id="id-14"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">bg_chrX = ballgown(dataDir = &#34;/home/username/RNA/chrX_data/result/ballgown&#34;, samplePattern = &#34;ERR&#34;, pData=pheno_data)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>滤去低丰度基因（样本间差异少于一个转录本）</li>
</ul>
<div class="highlight" id="id-15"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">bg_chrX_filt = subset(bg_chrX,&#34;rowVars(texpr(bg_chrX)) &gt;1&#34;,genomesubset=TRUE)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>得到组间有差异的转录本和基因</li>
</ul>
<div class="highlight" id="id-16"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">results_transcripts = stattest(bg_chrX_filt, feature=&#34;transcript&#34; covariate=&#34;sex&#34;,adjustvars = c(&#34;population&#34;),  getFC=TRUE, meas=&#34;FPKM&#34;)
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">results_genes = stattest(bg_chrX_filt, feature=&#34;gene&#34;, covariate=&#34;sex&#34;, adjustvars = c(&#34;population&#34;), getFC=TRUE, meas=&#34;FPKM&#34;)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>对转录本结果添加基因名</li>
</ul>
<div class="highlight" id="id-17"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">results_transcripts = data.frame(geneNames=ballgown::geneNames(bg_chrX_filt), geneIDs=ballgown::geneIDs(bg_chrX_filt), results_transcripts)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>转录本与基因结果按P值排序</li>
</ul>
<div class="highlight" id="id-18"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">results_transcripts=arrange(results_transcripts,pval)
</span></span><span class="line"><span class="cl">results_genes=arrange(results_genes,pval)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>将排序结果写入文件</li>
</ul>
<div class="highlight" id="id-19"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">write.csv(results_transcripts, &#34;chrX_transcript_results.csv&#34;,row。names=FALSE)
</span></span><span class="line"><span class="cl">write.csv(results_genes, &#34;chrX_gene_results.csv&#34;,row.names=FALSE)
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">#此步骤生成 chrX_transcript_results.csv, chrX_gene_results.csv
</span></span><span class="line"><span class="cl">#两个文字表格文件</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>识别q值小于0.05的转录物和基因（性别间表达有差异）</li>
</ul>
<div class="highlight" id="id-20"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">subset(results_transcripts,results_transcripts$qval&lt;0.05)
</span></span><span class="line"><span class="cl">subset(results_genes,results_genes$qval&lt;0.05)</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="结果可视化">结果可视化</h4>
<ul>
<li>FPKM为参考值，性别为区分条件作图</li>
</ul>
<div class="highlight" id="id-21"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">palette(rainbow(5))
</span></span><span class="line"><span class="cl">#为调色板指定颜色
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">fpkm = texpr(bg_chrX,meas=&#34;FPKM&#34;)
</span></span><span class="line"><span class="cl">fpkm = log2(fpkm+1)
</span></span><span class="line"><span class="cl">#提取FPKM值并做log转换
</span></span><span class="line"><span class="cl">boxplot(fpkm,col=as.numeric(pheno_data$sex),las=2,ylab=&#39;log2(FPKM+1)&#39;)</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>单个样本在转录组上的分布</li>
</ul>
<div class="highlight" id="id-22"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">ballgown::transcriptNames(bg_chrX)[12]
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">ballgown::geneNames(bg_chrX)[12]
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"># 绘制箱线图
</span></span><span class="line"><span class="cl">plot(fpkm[12,] ~ pheno_data$sex, border=c(1,2),
</span></span><span class="line"><span class="cl">    main=paste(ballgown::geneNames(bg_chrX)[12],&#39; : &#39;,
</span></span><span class="line"><span class="cl">    ballgown::transcriptNames(bg_chrX)[12]),pch=19, xlab=&#34;Sex&#34;,
</span></span><span class="line"><span class="cl">    ylab=&#39;log2(FPKM+1)&#39;)
</span></span><span class="line"><span class="cl">points(fpkm[12,] ~ jitter(as.numeric(pheno_data$sex)),
</span></span><span class="line"><span class="cl">    col=as.numeric(pheno_data$sex))</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>查看某一基因位置上所有转录组</li>
</ul>
<div class="highlight" id="id-23"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">plotTranscripts(ballgown::geneIDs(bg_chrX)[2263], bg_chrX, main=c(&#39;Gene XIST in sample ERR188234&#39;), sample=c(&#39;ERR188234&#39;))
</span></span><span class="line"><span class="cl">#根据ERR188234.gtf的文件，查出XIST基因所在位置编码2263，进行绘图</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>以性别为区分查看特定基因表达情况</li>
</ul>
<div class="highlight" id="id-24"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">plotMeans(&#39;MSTRG.56&#39;, bg_chrX_filt,groupvar=&#34;sex&#34;,legend=FALSE)</span></span></code></pre></td></tr></table>
</div>
</div><hr>
<p>由于作者精力有限，编写时难免会出现错误和过失，对您的阅读和操作造成的不便请您谅解。如果方便，还请把错误处邮件至本人邮箱，或于公众号处留言，十分感谢！</p>
</div><div class="post-reward">
    <div class="comment">Buy me a coffee~</div>
    <input type="checkbox" class="reward-input" name="reward" id="fi-reward" hidden />
    <label class="reward-button" for="fi-reward">赞赏</label>
    <div class="reward-ways" data-mode="fixed"><div><img loading="lazy" src="/images/alipay.jpg" srcset="/images/alipay.jpg, /images/alipay.jpg 1.5x, /images/alipay.jpg 2x" sizes="auto" data-title="flatig 支付宝" data-alt="flatig 支付宝" style="background: url(/svg/loading.min.svg) no-repeat center;" onload="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}this.dataset.lazyloaded='';" onerror="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}"/><span data-animation>支付宝</span>
          </div><div><img loading="lazy" src="/images/wechatpay.jpg" srcset="/images/wechatpay.jpg, /images/wechatpay.jpg 1.5x, /images/wechatpay.jpg 2x" sizes="auto" data-title="flatig 微信" data-alt="flatig 微信" style="background: url(/svg/loading.min.svg) no-repeat center;" onload="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}this.dataset.lazyloaded='';" onerror="this.title=this.dataset.title;this.alt=this.dataset.alt;for(const a of ['style','data-title','data-alt','onerror','onload']){this.removeAttribute(a);}"/><span data-animation>微信</span>
          </div></div>
  </div><div class="post-footer" id="post-footer">
  <div class="post-info">
    <div class="post-info-line">
      <div class="post-info-mod">
        <span title="更新于 2022-04-18 10:20:02">更新于 2022-04-18&nbsp;</span>
      </div><div class="post-info-license">
          <span><a rel="license external nofollow noopener noreferrer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span>
        </div></div>
    <div class="post-info-line">
      <div class="post-info-md"><span><a href="https://github.com/Flatigers/flatigers.github.io/edit/docs/content/posts/%e8%bf%90%e8%a1%8c%e4%b8%80%e4%b8%aaRNA-Seq%e6%b5%81%e7%a8%8b.md" title="编辑此页"target="_blank" rel="external nofollow noopener noreferrer" class="link-to-edit">编辑此页</a></span></div>
      <div class="post-info-share">
        <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/" data-title="运行一个RNA-Seq流程" data-hashtags="生物信息学,RNA-Seq"><i class="fa-brands fa-twitter fa-fw" aria-hidden="true"></i></a>
  <a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/" data-hashtag="生物信息学"><i class="fa-brands fa-facebook-square fa-fw" aria-hidden="true"></i></a>
  <a href="javascript:void(0);" title="分享到 Linkedin" data-sharer="linkedin" data-url="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/"><i class="fa-brands fa-linkedin fa-fw" aria-hidden="true"></i></a>
  <a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/" data-title="运行一个RNA-Seq流程"><i class="fa-brands fa-weibo fa-fw" aria-hidden="true"></i></a>
  <a href="javascript:void(0);" title="分享到 百度" data-sharer="baidu" data-url="https://flatig.vip/posts/%E8%BF%90%E8%A1%8C%E4%B8%80%E4%B8%AArna-seq%E6%B5%81%E7%A8%8B/" data-title="运行一个RNA-Seq流程"><i data-svg-src="/lib/simple-icons/icons/baidu.min.svg" aria-hidden="true"></i></a>
  </span>
      </div>
    </div>
  </div>

  <div class="post-info-more">
    <section class="post-tags"><i class="fa-solid fa-tags fa-fw me-1" aria-hidden="true"></i><a href='/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/' class="post-tag">生物信息学</a><a href='/tags/rna-seq/' class="post-tag">RNA-Seq</a></section>
    <section>
      <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
    </section>
  </div>

  <div class="post-nav"><a href="/posts/%E4%B8%AA%E4%BA%BA%E5%85%AC%E4%BC%97%E5%8F%B7%E5%BC%80%E9%80%9A%E5%95%A6/" class="post-nav-item" rel="prev" title="个人公众号开通啦"><i class="fa-solid fa-angle-left fa-fw" aria-hidden="true"></i>个人公众号开通啦</a>
      <a href="/posts/%E5%AE%89%E8%A3%85%E4%BD%BF%E7%94%A8%E6%9B%B4%E5%8F%AF%E9%9D%A0%E7%9A%84linux%E5%8F%91%E8%A1%8C%E7%89%88debian/" class="post-nav-item" rel="next" title="安装使用更可靠的Linux发行版—Debian">安装使用更可靠的Linux发行版—Debian<i class="fa-solid fa-angle-right fa-fw" aria-hidden="true"></i></a></div>
</div>
</article></main><footer class="footer">
    <div class="footer-container"><div class="footer-line powered">由 <a href="https://gohugo.io/" target="_blank" rel="external nofollow noopener noreferrer" title="Hugo 0.119.0">Hugo</a> 强力驱动 | 主题 - <a href="https://github.com/hugo-fixit/FixIt" target="_blank" rel="external" title="FixIt v0.2.18-lts.1"><img class="fixit-icon" src="/fixit.min.svg" alt="FixIt logo" />&nbsp;FixIt</a>
        </div><div class="footer-line copyright" itemscope itemtype="http://schema.org/CreativeWork"><i class="fa-regular fa-copyright fa-fw" aria-hidden="true"></i>
            <span itemprop="copyrightYear">2023</span><span class="author" itemprop="copyrightHolder">
              <a href="https://github.com/Flatigers"target="_blank" rel="external nofollow noopener noreferrer">flatig</a></span><span class="license footer-divider"><a rel="license external nofollow noopener noreferrer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span></div><div class="footer-line statistics order-first"><span class="site-time" title='网站运行中……'><i class="fa-solid fa-heartbeat fa-fw animate-icon" aria-hidden="true"></i><span class="ms-1 d-none">博客已运行</span><span class="run-times ms-1">网站运行中……</span></span></div></div>
  </footer></div><div class="widgets"><div class="fixed-buttons animate__faster d-none"><div class="fixed-button back-to-top" role="button" aria-label="回到顶部"><i class="fa-solid fa-arrow-up fa-fw" aria-hidden="true"></i><span class="variant-numeric">0%</span>
        </div></div><a href="https://github.com/Flatigers" title="我的GitHub，看看有没有你感兴趣的项目"target="_blank" rel="external nofollow" class="github-corner right d-none-mobile"><svg viewBox="0 0 250 250" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><div id="mask"></div><div class="reading-progress-bar" style="left: 0;top: 0;--bg-progress: #000;--bg-progress-dark: #fff;"></div><noscript>
    <div class="noscript-warning">FixIt 主题在启用 JavaScript 的情况下效果最佳。</div>
  </noscript>
</div><link rel="stylesheet" href="/lib/lightgallery/css/lightgallery-bundle.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/cookieconsent/cookieconsent.min.css"><link rel="stylesheet" href="/lib/pace/themes/blue/pace-theme-minimal.css"><script src="/lib/autocomplete/autocomplete.min.js" defer></script><script src="/lib/algoliasearch/algoliasearch-lite.umd.min.js" defer></script><script src="/lib/instant-page/instantpage.min.js" async defer type="module"></script><script src="/lib/twemoji/twemoji.min.js" defer></script><script src="/lib/lightgallery/lightgallery.min.js" defer></script><script src="/lib/lightgallery/plugins/thumbnail/lg-thumbnail.min.js" defer></script><script src="/lib/lightgallery/plugins/zoom/lg-zoom.min.js" defer></script><script src="/lib/sharer/sharer.min.js" async defer></script><script src="/lib/typeit/index.umd.js" defer></script><script src="/lib/katex/katex.min.js" defer></script><script src="/lib/katex/auto-render.min.js" defer></script><script src="/lib/katex/copy-tex.min.js" defer></script><script src="/lib/katex/mhchem.min.js" defer></script><script src="/lib/cookieconsent/cookieconsent.min.js" defer></script><script src="/lib/pangu/pangu.min.js" defer></script><script src="/lib/cell-watermark/watermark.min.js" defer></script><script src="/lib/pace/pace.min.js" async defer></script><script>window.config={"autoBookmark":true,"code":{"copyTitle":"复制到剪贴板","editLockTitle":"锁定可编辑代码块","editUnLockTitle":"解锁可编辑代码块","editable":true,"maxShownLines":50},"comment":{"enable":false},"cookieconsent":{"content":{"dismiss":"同意","link":"了解更多","message":"本网站使用 Cookies 来改善您的浏览体验。"},"enable":true,"palette":{"button":{"background":"#f0f0f0"},"popup":{"background":"#1aa3ff"}},"theme":"edgeless"},"data":{"typeit-header-subtitle-desktop":"博客|技术分享","typeit-header-subtitle-mobile":"博客|技术分享"},"enablePWA":true,"lightgallery":true,"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":true,"left":"\\begin{equation}","right":"\\end{equation}"},{"display":true,"left":"\\begin{equation*}","right":"\\end{equation*}"},{"display":true,"left":"\\begin{align}","right":"\\end{align}"},{"display":true,"left":"\\begin{align*}","right":"\\end{align*}"},{"display":true,"left":"\\begin{alignat}","right":"\\end{alignat}"},{"display":true,"left":"\\begin{alignat*}","right":"\\end{alignat*}"},{"display":true,"left":"\\begin{gather}","right":"\\end{gather}"},{"display":true,"left":"\\begin{CD}","right":"\\end{CD}"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"pangu":{"enable":true,"selector":"article"},"search":{"algoliaAppID":"","algoliaIndex":"","algoliaSearchKey":"","highlightTag":"em","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"algolia"},"siteTime":"2023-10-14T11:30:00+08:00","twemoji":true,"typeit":{"cursorChar":"|","cursorSpeed":1000,"data":{"typeit-header-subtitle-desktop":["typeit-header-subtitle-desktop"],"typeit-header-subtitle-mobile":["typeit-header-subtitle-mobile"]},"duration":-1,"loop":false,"speed":100},"watermark":{"appendto":".wrapper\u003emain","colspacing":30,"content":"\u003cimg style=\"height: 0.85rem;\" src=\"/logo.png\" alt=\"logo\" /\u003e flatig","enable":true,"fontfamily":"MMT_LRH,沐目体","fontsize":1.1,"height":20,"opacity":0.0125,"rotate":15,"rowspacing":60,"width":150}};</script><script src="/js/theme.min.js" defer></script></body>
</html>
